components/dashboard/src/onboarding/OnboardingStep.tsx

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { FC, FormEvent, useCallback } from "react";
+import Alert from "../components/Alert";
+
+type Props = {
+    title: string;
+    subtitle: string;
+    isValid: boolean;
+    isLoading?: boolean;
+    error?: string;
+    onSubmit(): void;
+};
```

Comment:
AFAIK these props are for Onboarding Forms.

`isLoading` looks ambiguous as we are already using in List Workspace ([code](https://github.com/gitpod-io/gitpod/blob/f2d868e7eaa111bf6308cb7c4acc085e7c3775e7/components/dashboard/src/workspaces/Workspaces.tsx#L30)). e.g. `isFormLoading`could be good. Or for quick :shipit: we can just add the comment for context.


```suggestion
type Props = {
  // The title of the form
  title: string;
  // The subtitle of the form
  subtitle: string;
  // Whether the form is valid
  isValid: boolean;
  // Whether the form is loading
  isLoading?: boolean;
  // A message to display if there is an error
  error?: string;
  // A function to call when the form is submitted
  onSubmit(): void;
};
```

Reply:
Good point, the more I think about it, `isLoading` isn't very indicative of what's happening here, it's really indicating the form is saving, and puts it in a disabled state. I'll adjust it to something like `isSaving` - and probably need to add another check to avoid redundant submits if that flag is set.